### PR TITLE
Migration to sqlalchemy

### PR DESCRIPTION
**Pull request by shaddygarg**
#### Checklist
- [x] I have followed the [Contributor Guidelines](https://github.com/zdresearch/OWASP-Nettacker/wiki/Developers#contribution-guidelines).
- [x] I have added the relevant documentation.
- [x] My branch is up-to-date with the Upstream master branch.

#### Changes proposed in this pull request
This is a work in progress pull request for the migration of the current db to sqlalchemy. This PR is not to be merged yet. This is in reference to the #138 .
#### Work to be done:
-  ~~Wrote the initial database structure for the project~~
- ~~Changed the database structure of the project~~
- ~~Tested the code by porting it to MySQL and sqlite3~~
- ~~Test all the functions for these two databases~~
- ~~Implement a structure similar to the first answer of https://stackoverflow.com/questions/15275799/sqlalchemy-and-multiple-databases~~
- ~~Update the requirements.txt~~
- ~~Set the initial creation setup~~
-  Write the documentation for the new database setup
#### Your development environment
- OS: `Ubuntu`
- OS Version: `16.04`
- Python Version: `2.7`

### CODE REVIEW

File: database/config.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+DB = 'sqlite'
```
Comment (Ali-Razmjoo) [inline database/config.py]: please add the standard header on top of this file.
```python
#!/usr/bin/env python
# -*- coding: utf-8 -*-
```

File: database/config.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+#USER = ''
+#PASSWORD = ''
+#HOST = ''
+#PORT = ''
+DATABASE = 'nettacker.db'
```
Comment (Ali-Razmjoo) [inline database/config.py]: Please insert database values in json/dict in [core/config.py](https://github.com/zdresearch/OWASP-Nettacker/blob/master/core/config.py)  with default values. then import the config.py in here and use it to read the config. e.g.
```python
from core.config import _database_config
USER = _database_config()["USERNAME"]
```

File: database/models.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy import Column, Integer, Text
+
+Base = declarative_base()
+
+
+class Report(Base):
```
Comment (Ali-Razmjoo) [inline database/models.py]: add code-based documentation for the class.
class Report(Base):
    """
    Class description
    """

File: database/config.py (hunk 2 above)
Comment (shaddygarg) [reply database/config.py]: Sure, I will do that.

File: database/models.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    scan_method = Column(Text)
+    language = Column(Text)
+    scan_cmd = Column(Text)
+    ports = Column(Text)
+
+    def __repr__(self):
+        return "<Report(id='%s', scan_id='%s', date='%s')>" % (self.id, self.scan_id, self.date)
+
+
+class HostsLog(Base):
```
Comment (Ali-Razmjoo) [inline database/models.py]: add code-based documentation for class

File: database/mysql_create.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+import time
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+from database.config import HOST, USER, PASSWORD, PORT, DATABASE
+from database.models import Base
+from core.alert import warn
+from core.alert import messages
+
+
+def create_database():
```
Comment (Ali-Razmjoo) [inline database/mysql_create.py]: add code-based documentation for this function. specify the description and return.

File: database/mysql_create.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        existing_databases = engine.execute("SHOW DATABASES;")
+        existing_databases = [d[0] for d in existing_databases]
+        if DATABASE not in existing_databases:
+            engine.execute("CREATE DATABASE {0} ".format(DATABASE))
+        return True
+    except:
+        return False
+
+
+def create_tables():
```
Comment (Ali-Razmjoo) [inline database/mysql_create.py]: add code-based documentation for this function. specify the description and return.

File: database/models.py (hunk 3 above)
Comment (shaddygarg) [reply database/models.py]: Sure, this is a remaining task in this PR

File: database/mysql_create.py (hunk 5 above)
Comment (shaddygarg) [reply database/mysql_create.py]: Sure, I will. This is a pending task that I will complete soon.

**Comment by shaddygarg** [comment]
@Ali-Razmjoo please review the PR. I have completed the changes.

**Comment by Ali-Razmjoo** [comment]
Hello,

Thanks for your contribution, PR merged, I will mark #138 as done!

Regards.

**Pull request merged**